server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java

```diff
@@ -118,7 +118,7 @@ public Session generateSession(DataReadRequest request) {
     @Transactional
     public DataReadLog assignmentData(String consumerId, Session session) {
         var sid = session.getId();
-        var queue = sessionCache.computeIfAbsent(String.valueOf(sid), id -> new ConcurrentLinkedQueue<>());
+        var queue = sessionCache.computeIfAbsent(String.valueOf(sid), id -> new LinkedList<>());
 
         if (queue.isEmpty()) {
             queue.addAll(dataReadLogDao.selectTopsUnAssignedData(sid, cacheSize));
```

Comment:
Release the key if empty?

Reply:
done, use weakHashMap, auto released by GC